Source/CollisionConnect.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <QUndoCommand>
+#include "Model.hpp"
+#include "ResizeableArrowItem.hpp"
+
+struct CollisionConnectData
+{
+    CollisionConnectData(ObjectProperty *mObjectProperty, int mOldValue, int mNewValue) : mObjectProperty(
+            mObjectProperty), mOldValue(mOldValue), mNewValue(mNewValue)
+    {
+
+    }
+
+    ObjectProperty* mObjectProperty;
+    int mOldValue;
+    int mNewValue;
+};
+
+class CollisionConnectCommand final : public QUndoCommand
+{
+public:
+    CollisionConnectCommand(std::vector<CollisionConnectData> collisionConnectData);
+
+    void undo() override;
+
+    void redo() override;
+
+private:
+    std::vector<CollisionConnectData> mCollisionConnectData;
+
+};
+
+std::vector<CollisionConnectData> getConnectCollisionsChanges(const std::vector<ResizeableArrowItem *> &collisions);
```

Comment:
this can go in the cpp only since nothing calls it externally? and then it can also be marked file scope/static

Reply:
moved it into ``CollisionConnectCommand`` and made static